.. _summary:

*******
Summary
*******

Proteins are generally comprised of one or more functional regions, commonly termed domains. The presence of different domains in varying combinations in different proteins gives rise to the diverse repertoire of proteins found in nature. Identifying the domains present in a protein can provide insights into the function of that protein.

The Pfam database is a large collection of protein domain families. Each family is represented by multiple sequence alignments and a profile hidden Markov model (HMM).

Each Pfam family, sometimes referred to as a Pfam-A entry, consists of a curated seed alignment containing a small set of representative members of the family, profile HMMs built from the seed alignment, and an automatically generated full alignment, which contains all detectable protein sequences belonging to the family, as defined by profile HMM searches of primary sequence databases.

Pfam entries are classified in one of six types:

**Family**
  A collection of related protein regions

**Domain**
  A structural unit

**Repeat**
  A short unit which is unstable in isolation but forms a stable structure when multiple copies are present

**Motifs**
  A short unit found outside globular domains

**Coiled-Coil**
  Regions that predominantly contain coiled-coil motifs, regions that typically contain alpha-helices that are coiled together in bundles of 2-7.

**Disordered**
  Regions that are conserved, yet are either shown or predicted to contain bias sequence composition and/or are intrinsically disordered (non-globular).

Related Pfam entries are grouped together into clans; the relationship may be defined by similarity of sequence, structure or profile HMM. 
